runtime/src/bank.rs

```diff
@@ -4397,6 +4408,7 @@ impl Bank {
                                 !self
                                     .feature_set
                                     .is_active(&remove_deprecated_request_unit_ix::id()),
+                                true, // don't reject txs that use request heap size ix
```

Comment:
This bit is nasty. It has to be `true` to stay in sync with v1.13 [here](https://github.com/solana-labs/solana/blob/v1.13.6/runtime/src/bank.rs#LL4181C33-L4181C85). Since `requestable_heap_size::id()` had been cleared, I didn't re-introduce it back, but use `true` in same fashion as [this](https://github.com/solana-labs/solana/blob/v1.13.6/core/src/unprocessed_packet_batches.rs#L419)

Early I passed `self.enable_request_heap_frame_ix()`, which is `false` when replay with mainnet-beta snapshot. The transaction errored out with `InstructionError::InvalidInstructionData`, bank `175475896` didn't root. (Glad re-ran it on same ledger, good call @steviez!)

With this change, replay produced correct hash: 
```
[2023-02-03T04:57:11.929764473Z INFO  solana_ledger_tool] solana-ledger-tool 1.16.0 (src:00000000; feat:1471661895, client:SolanaLabs)
... ...
[2023-02-03T05:01:10.816485948Z INFO  solana_runtime::bank] bank frozen: 175475896 hash: GSgpmbzC21XqNJmdNd84MQwjfn5njpeyvSSPo1RFRyHY ... ...
```

Reply:
Dang, nice, glad we did one more check so we didn't have to roll out v1.14 versions in subsequent days